packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -570,6 +571,30 @@ class ReleaseIosApplicationBundle extends IosAssetBundle {
   List<Target> get dependencies => const <Target>[
     AotAssemblyRelease(),
   ];
+
+  @override
+  Future<void> build(Environment environment) async {
+    bool buildSuccess = true;
+    try {
+      await super.build(environment);
+    } catch (_) {  // ignore: avoid_catches_without_on_clauses
+      buildSuccess = false;
+      rethrow;
+    } finally {
+      // Send a usage event when the app is being archived.
+      // Since assemble is run during a `flutter build`/`run` as well as an out-of-band
+      // archive command from Xcode, this is a more accurate count than `flutter build ipa` alone.
+      if (environment.defines[kXcodeAction]?.toLowerCase() == 'install') {
+        environment.logger.printTrace('Sending archive event if usage enabled.');
+        UsageEvent(
+          'assemble',
+          'ios-archive',
```

Comment:
Should we have an analogous event on the Android (, desktop, web, ...) side?

Reply:
Does `flutter build apk` already fully cover Android? Or does it have the same issue where it can be bundled and released without `flutter`?  Web seems to say "upload a release version" which means the tool may not know if the app is being "published" or just being built in release mode.   Windows and Linux both have publication instructions outside of the tool, so that may benefit from a similar pattern to this, if `assemble` is called, and there's some way to know it's being packaged.  Each platform would need individual investigation.